lib/wpMandrill.class.php

```diff
@@ -40,14 +40,19 @@ function wp_mail( $to, $subject, $message, $headers = '', $attachments = array()
                 try {
                     $sent = wpMandrill::mail( $to, $subject, $message, $headers, $attachments );
 
-                    if (    is_wp_error($sent)
-                        || !isset($sent[0]['status'])
-                        || ($sent[0]['status'] != 'sent' && $sent[0]['status'] != 'queued') ) {
+                    if (
+                        is_wp_error( $sent )
+                        || !isset( $sent[0]['status'] )
+                        || in_array( $sent[0]['status'], array( 'sent', 'queued' ) )
+                        || (
+                            'rejected' === $sent[0]['status']
+                            && isset( $sent[0]['reject_reason'] )
+                            && 'hard-bounce' === $sent[0]['reject_reason']
+                        )
```

Comment:
Ugh, I think you're right (reviewing from mobile); I'll fix and work on the abstraction as well.

Reply:
Ok great, thanks! I think it's best to make those adjustments now rather than later so we don't fall into the same trap that the original developers did.

Reply:
@MillerMedia Fixed the logic, and moved into separate function, `evaluate_response()`.

Note that I also included `error_log()` from #20, as the `master` branch (what I branched from) does not yet include that functionality.